api/proto/storage/s3.proto

```diff
@@ -0,0 +1,50 @@
+syntax = "proto3";
+package proto.storage;
+
+option go_package = "github.com/spinnaker/kleat/api/client/storage";
+
+// Configuration for an Amazon S3 persistent store.
+message S3 {
+  // Whether this persistent store is enabled.
+  string enabled = 1;
+
+  // The name of a storage bucket that your specified account has access to.
+  string bucket = 2;
+
+  // The root folder in the chosen bucket to place all of Spinnaker’s persistent
+  // data in.
+  string rootFolder = 3;
+
+  // This is only required if the bucket you specify doesn’t exist yet. In that
+  // case, the bucket will be created in that region.
```

Comment:
Was it Halyard creating the bucket? (Should we remove the "this is only required..." part?)

Reply:
Not, it's actually front50 that's [creating it](https://github.com/spinnaker/front50/blob/d4d33ed34911a5e36fb67d1ee63e5c659a6b3b55/front50-s3/src/main/java/com/netflix/spinnaker/front50/model/S3StorageService.java#L73) so we should probably leave it.

The fact that Halyard was picking a bucket name was confusing and hard to support (as kleat doesn't write back to the config so would have no place to put the generated name, and would make a new one every time).  But this seems reasonable as at least the config stays the same every time.